src/Frontend/state/actions/popup-actions/popup-actions.ts

```diff
@@ -119,9 +144,49 @@ export function navigateToTargetResourceOrAttributionOrOpenFileDialog(): AppThun
 export function closePopupAndUnsetTargets(): AppThunkAction {
   return (dispatch) => {
     dispatch(setTargetView(null));
-    dispatch(setTargetSelectedResourceId(''));
-    dispatch(setTargetSelectedAttributionId(''));
+    dispatch(setTargetSelectedResourceId(null));
+    dispatch(setTargetSelectedAttributionId(null));
     dispatch(closePopup());
     dispatch(setOpenFileRequest(false));
+    dispatch(setImportFileRequest(null));
+    dispatch(setExportFileRequest(null));
+  };
+}
+
+export function showImportDialogWithUnsavedCheck(
+  fileFormat: FileFormatInfo,
+): AppThunkAction {
+  return (dispatch, _) => {
+    dispatch(
+      withUnsavedCheck(
+        () =>
+          dispatch(openPopup(PopupType.ImportDialog, undefined, fileFormat)),
+        () => dispatch(setImportFileRequest(fileFormat)),
+      ),
+    );
+  };
+}
```

Comment:
something weird is going on here: you're wrapping `withUnsavedCheck` in a dispatch but then again calling `dispatch` on `showImportDialogWithUnsavedCheck` where you're using it. this is not how you use, e.g., `navigateToSelectedPathOrOpenUnsavedPopup` which is not wrapped in a dispatch.
so shouldn't this just be:
```suggestion
  return withUnsavedCheck(
        (dispatch) =>
          dispatch(openPopup(PopupType.ImportDialog, undefined, fileFormat)),
        (dispatch) => dispatch(setImportFileRequest(fileFormat)),
      );
}
```
same questions about the actions below.